src/libraries/System.Text.Json/tests/Serialization/CollectionTests/CollectionTests.Generic.Read.cs

```diff
@@ -257,6 +257,28 @@ public static void ReadSimpleGenericIList()
             Assert.Equal(0, result2.Count());
         }
 
+        [Fact]
+        public static void ReadGenericStructIList()
+        {
+            string json = "[10,20,30]";
+            var wrapper = JsonSerializer.Deserialize<GenericStructIListWrapper<int>>(json);
```

Comment:
Can you also test that deserializing a nullable struct collection type works?

`JsonSerializer.Deserialize<GenericStructIListWrapper<int>?>(json);`
`JsonSerializer.Deserialize<GenericStructIListWrapper<int>?>("null");`